### PR TITLE
Ingm 476 make fsoe master module optional

### PR DESCRIPTION
**Pull request by GullinBustin**
Fixes # INGM-476

### Type of change

- [x] Add fsoe_master as extra requirement
- [x] Turn on python 3.12 test again

### Tests
- [x] Add new unit tests if it applies.
- [x] Run tests.

### Documentation

Please update the documentation.

- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use the ruff package to format the code: `ruff format ingeniamotion tests`.
- [x] Use the ruff package to lint the code: `ruff check ingeniamotion`.

### Others

- [x] Set fix version field in the Jira issue.

### CODE REVIEW

File: ingeniamotion/motion_controller.py [hunk 1]
```diff
@@ -146,6 +153,8 @@ def info(self) -> Information:
     @property
     def fsoe(self) -> "FSoEMaster":
         """Instance of :class:`~ingeniamotion.fsoe.FSoEMaster` class"""
+        if not FSOE_MASTER_INSTALLED:
+            raise NotImplementedError("The FSoE module is not available.")
```
Comment (polfeliu) [inline ingeniamotion/motion_controller.py:157]: I would indicate that the package must be installed with the extra specified:
`pip install ingeniamotion[FSoE]`

File: ingeniamotion/motion_controller.py [hunk 2]
```diff
@@ -9,12 +9,18 @@
...
-from ingeniamotion.fsoe import FSoEMaster
 from ingeniamotion.information import Information
 from ingeniamotion.io import InputsOutputs
 from ingeniamotion.metaclass import DEFAULT_AXIS, DEFAULT_SERVO
 from ingeniamotion.motion import Motion
 
+try:
+    from ingeniamotion.fsoe import FSoEMaster
+except ImportError:
+    FSOE_MASTER_INSTALLED = False
```
Comment (polfeliu) [inline ingeniamotion/motion_controller.py:20]: I would place this on the import of fsoe-master instead of the ingeniamotion to avoid masking other errors

**Review by MartinALL91 — APPROVED** [review]

File: ingeniamotion/motion_controller.py (hunk 1 above)
Comment (GullinBustin) [reply ingeniamotion/motion_controller.py]: d23549a

File: ingeniamotion/motion_controller.py (hunk 2 above)
Comment (GullinBustin) [reply ingeniamotion/motion_controller.py]: d23549a

**Review by polfeliu — APPROVED** [review]

**Pull request merged**